dynawo/sources/Modeler/DataInterface/DYNDataInterfaceIIDMCommon.cpp

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright (c) 2022, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+#include "DYNDataInterfaceIIDM.h"
+
+#include "DYNNetworkInterface.h"
+#include "DYNBusInterface.h"
+#include "DYNSwitchInterface.h"
+#include "DYNLoadInterface.h"
+#include "DYNLineInterface.h"
+#include "DYNGeneratorInterface.h"
+#include "DYNShuntCompensatorInterface.h"
+#include "DYNStaticVarCompensatorInterface.h"
+#include "DYNDanglingLineInterface.h"
+#include "DYNTwoWTransformerInterface.h"
+#include "DYNThreeWTransformerInterface.h"
+#include "DYNHvdcLineInterface.h"
+#include "DYNVoltageLevelInterface.h"
+
+using boost::shared_ptr;
+using std::vector;
+
+namespace DYN {
+
+bool
+DataInterfaceIIDM::instantiateNetwork() const {
+  shared_ptr<NetworkInterface> network = getNetwork();
+  const vector<shared_ptr<VoltageLevelInterface> >& voltageLevels = network->getVoltageLevels();
+  vector<shared_ptr<VoltageLevelInterface> >::const_iterator iVL;
```

Comment:
Please insert the iterator declaration Inside the for loop (same everywhere)

Reply:
done

Reply:
I refactored the code entirely